src/rules/jsxCurlySpacingRule.ts

```diff
@@ -125,36 +122,76 @@ class JsxCurlySpacingWalker extends Lint.RuleWalker {
         const nodeWidth = node.getWidth();
 
         if (this.hasOption(OPTION_ALWAYS)) {
-            if (!isSpaceBetweenTokens(firstToken, secondToken)) {
+            if (!this.getDeleteFixForSpaceBetweenTokens(firstToken, secondToken)) {
+                const fix = new Lint.Fix(Rule.metadata.ruleName, [
+                    this.appendText(secondToken.getFullStart(), " "),
+                ]);
                 let failureString = Rule.FAILURE_NO_BEGINNING_SPACE(firstToken.getText());
 
-                this.addFailure(this.createFailure(nodeStart, 1, failureString));
+                this.addFailure(this.createFailure(nodeStart, 1, failureString, fix));
             }
 
-            if (!isSpaceBetweenTokens(secondToLastToken, lastToken)) {
+            if (!this.getDeleteFixForSpaceBetweenTokens(secondToLastToken, lastToken)) {
+                const fix = new Lint.Fix(Rule.metadata.ruleName, [
+                    this.appendText(lastToken.getStart(), " "),
+                ]);
                 let failureString = Rule.FAILURE_NO_ENDING_SPACE(lastToken.getText());
 
-                this.addFailure(this.createFailure(nodeStart +  nodeWidth - 1, 1, failureString));
+                this.addFailure(this.createFailure(nodeStart + nodeWidth - 1, 1, failureString, fix));
             }
         } else if (this.hasOption(OPTION_NEVER)) {
             const firstAndSecondTokensCombinedText = getTokensCombinedText(firstToken, secondToken);
             const lastAndSecondToLastCombinedText = getTokensCombinedText(secondToLastToken, lastToken);
 
             if (!isExpressionMultiline(firstAndSecondTokensCombinedText)) {
-                if (isSpaceBetweenTokens(firstToken, secondToken)) {
+                let fix = this.getDeleteFixForSpaceBetweenTokens(firstToken, secondToken);
+                if (fix) {
                     let failureString = Rule.FAILURE_FORBIDDEN_SPACES_BEGINNING(firstToken.getText());
 
-                    this.addFailure(this.createFailure(nodeStart, 1, failureString));
+                    this.addFailure(this.createFailure(nodeStart, 1, failureString, fix));
                 }
             }
 
             if (!isExpressionMultiline(lastAndSecondToLastCombinedText)) {
-                if (isSpaceBetweenTokens(secondToLastToken, lastToken)) {
+                let fix = this.getDeleteFixForSpaceBetweenTokens(secondToLastToken, lastToken);
+                if (fix) {
```

Comment:
no implicit boolean conversion -- after applying the fix below, make this `if (fix !== undefined)`